KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java

```diff
@@ -491,6 +500,19 @@ public void onEvent(IKalturaEvent event){
 		}
 	}
 
+	protected  void onStreamReadyForPlayback(String entryId, final KalturaMediaServerIndex serverIndex) {
+		logger.debug("onStreamReadyForPlayback [" + entryId + "]");
+
+		synchronized (entries) {
```

Comment:
I think here is your bug 

onStreamReadyForPlayback which is called from the ChunkIdHandler code,
but you lock the entries object which in the kaltura code can be locked for a long time (API calls)
this causes delays in locations you cannot have,
+ setEntryIsLive is synchornized, meaning it will lock for more:
here is what I suggest to do:
1. run onStreamReadyForPlayback in a timer ( + add ref. to this timer + delete timer after it's firied)
2. delete the TimerTask in the setEntryIsLive (since we call it from a timer anyways)